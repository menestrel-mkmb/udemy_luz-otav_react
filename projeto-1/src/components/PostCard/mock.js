export const postCardMockProps = {
    id: 1,
    title: 'TÃ­tulo 1',
    body: 'Lorem1',
    url: 'img/img.jpg'
};