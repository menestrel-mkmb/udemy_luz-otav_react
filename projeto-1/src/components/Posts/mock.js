export const postsMockProps = [
    {
        id: 1,
        title: 'Título 1',
        body: 'Lorem1',
        url: 'img/img1.jpg'
    },
    {
        id: 2,
        title: 'Título 2',
        body: 'Lorem2',
        url: 'img/img2.jpg'
    },
    {
        id: 3,
        title: 'Título 3',
        body: 'Lorem3',
        url: 'img/img3.jpg'
    }
];