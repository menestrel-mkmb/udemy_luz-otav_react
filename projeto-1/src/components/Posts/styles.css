.posts {
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    gap: .5rem;

    margin: 0 auto;
    padding: 1rem;
}

.search-result__text {
    font-size: .75rem;
}

.search-result__text em {
    font-weight: 700;
}

.wrapper {
    align-self: stretch;

    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: 1rem;
  
    padding: 1rem;
}