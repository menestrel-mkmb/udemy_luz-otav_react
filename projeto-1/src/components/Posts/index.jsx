import './styles.css';

import PostCard from '../PostCard';

const Posts = (props) => {
  const { firstLoad, posts } = props;

  return (
    <article className='posts wrapper'>

      { posts.length > 0 ?
        posts.map(
          post => <PostCard post={post} key={post.id} />
        ) : (
          <section
            className='wrapper'
          >
            { firstLoad ?
              <p>('Carregando...')</p> :
              <p>('NaÌƒo existem resultados para essa busca)</p>
            }
          </section>
        )
      }
    </article>
  );
}

export default Posts;
