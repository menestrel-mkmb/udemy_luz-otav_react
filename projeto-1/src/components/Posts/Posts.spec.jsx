import { render, screen } from '@testing-library/react';
import Posts from '.';
import { postsMockProps } from './mock';

const props = postsMockProps;

describe('<Posts />', () => {
    it('should render Posts', () => {
        render(<Posts posts={props} />);

        expect(screen.getAllByRole('heading', { name: /título/i }))
            .toHaveLength(postsMockProps.length);
        expect(screen.getAllByRole('img', { name: /título/i }))
            .toHaveLength(postsMockProps.length);
        expect(screen.getAllByText( /Lorem/i ))
            .toHaveLength(postsMockProps.length);
        expect(screen.getByRole('img', { name: /título 3/i }))
            .toHaveAttribute('src', 'img/img3.jpg');
    });

    it('should render no posts', () => {
        render(<Posts posts={[]} />);
        expect(screen.queryByRole('heading', { name: /título/i }))
            .not.toBeInTheDocument();
    })

    it('should match snapshot', () => {
        const { container } = render(<Posts posts={props} />);
        expect(container.firstChild).toMatchSnapshot();
    })
});