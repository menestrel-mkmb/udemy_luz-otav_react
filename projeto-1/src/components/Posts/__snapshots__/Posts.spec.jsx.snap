// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<article
  class="posts wrapper"
>
  <div
    class="post card"
  >
    <img
      alt="Título 1"
      class="post__img"
      src="img/img1.jpg"
    />
    <div
      class="post__content"
    >
      <h1
        class="post__title"
      >
        Título 1
      </h1>
      <p
        class="post__body"
      >
        Lorem1
      </p>
    </div>
  </div>
  <div
    class="post card"
  >
    <img
      alt="Título 2"
      class="post__img"
      src="img/img2.jpg"
    />
    <div
      class="post__content"
    >
      <h1
        class="post__title"
      >
        Título 2
      </h1>
      <p
        class="post__body"
      >
        Lorem2
      </p>
    </div>
  </div>
  <div
    class="post card"
  >
    <img
      alt="Título 3"
      class="post__img"
      src="img/img3.jpg"
    />
    <div
      class="post__content"
    >
      <h1
        class="post__title"
      >
        Título 3
      </h1>
      <p
        class="post__body"
      >
        Lorem3
      </p>
    </div>
  </div>
</article>
`;
